From: Arnd Bergmann <arnd@arndb.de>
Subject: syscalls: add sys_ni_posix_timers prototype
Date: Wed, 7 Jun 2023 16:28:45 +0200

The sys_ni_posix_timers() definition causes a warning when the declaration
is missing, so this needs to be added along with the normal syscalls,
outside of the #ifdef.

kernel/time/posix-stubs.c:26:17: error: no previous prototype for 'sys_ni_posix_timers' [-Werror=missing-prototypes]

Link: https://lkml.kernel.org/r/20230607142925.3126422-1-arnd@kernel.org
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/alpha/kernel/osf_sys.c |    2 --
 include/linux/syscalls.h    |    1 +
 kernel/time/posix-stubs.c   |    1 +
 3 files changed, 2 insertions(+), 2 deletions(-)

--- a/arch/alpha/kernel/osf_sys.c~syscalls-add-sys_ni_posix_timers-prototype
+++ a/arch/alpha/kernel/osf_sys.c
@@ -1014,8 +1014,6 @@ SYSCALL_DEFINE2(osf_settimeofday, struct
 	return do_sys_settimeofday64(tv ? &kts : NULL, tz ? &ktz : NULL);
 }
 
-asmlinkage long sys_ni_posix_timers(void);
-
 SYSCALL_DEFINE2(osf_utimes, const char __user *, filename,
 		struct timeval32 __user *, tvs)
 {
--- a/include/linux/syscalls.h~syscalls-add-sys_ni_posix_timers-prototype
+++ a/include/linux/syscalls.h
@@ -1285,6 +1285,7 @@ asmlinkage long sys_ni_syscall(void);
 
 #endif /* CONFIG_ARCH_HAS_SYSCALL_WRAPPER */
 
+asmlinkage long sys_ni_posix_timers(void);
 
 /*
  * Kernel code should not call syscalls (i.e., sys_xyzyyz()) directly.
--- a/kernel/time/posix-stubs.c~syscalls-add-sys_ni_posix_timers-prototype
+++ a/kernel/time/posix-stubs.c
@@ -16,6 +16,7 @@
 #include <linux/posix-timers.h>
 #include <linux/time_namespace.h>
 #include <linux/compat.h>
+#include <linux/syscalls.h>
 
 #ifdef CONFIG_ARCH_HAS_SYSCALL_WRAPPER
 /* Architectures may override SYS_NI and COMPAT_SYS_NI */
_
